tests/queries/0_stateless/03167_parametrized_view_with_cte.sql

```diff
@@ -0,0 +1,4 @@
+SET allow_experimental_analyzer=1;
+CREATE OR REPLACE VIEW param_test AS SELECT {test_str:String} as s_result;
+WITH 'OK' AS s SELECT * FROM param_test(test_str=s);
+WITH CAST(123, String) AS s SELECT * FROM param_test(test_str=s);
```

Comment:
Can we also make inline expressions work, such as `test = 1 + 2` or `test = (SELECT 1)`?

Reply:
This part is easier to support.

Reply:
> Can we also make inline expressions work, such as test = 1 + 2 or test = (SELECT 1)?

Done.

Reply:
Let's also add tests for such cases.

Reply:
> Let's also add tests for such cases.

But I did add...